### PR TITLE
Add Threads v0.4.0

### PR DESCRIPTION
**Pull request by ed-brex**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

---

* The version is `0.X`; however, this is the officially provided one from https://threads.com
* First formula contribution - hope I covered/searched everything correctly in the provided links
    * Also tried to discover the files/dirs it creates for `zap`. Wasn't 100% sure of `trash` vs `rmdir`

### CODE REVIEW

File: Casks/threads.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+cask 'threads' do
+  version '0.4.0'
+  sha256 '025669c445510e1ba4511a3bc51dcec2454fca4ce9dfb30c0b3fff95afc18d4b'
+
+  url "https://downloads.threads.com/mac/Threads-#{version}.dmg"
```
Comment (vitorgalvao) [inline Casks/threads.rb]: Is there a public facing page where one can access this download? I’m trying to find an [`appcast`](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md).

Comment (ed-flanagan) [reply Casks/threads.rb]: Nothing that I saw immediately. I can poke around a little. I believe this is a relatively new release for them. I can try emailing their support, see if they have a feed or something.

Comment (vitorgalvao) [reply Casks/threads.rb]: > I can try emailing their support, see if they have a feed or something.

That would be appreciated.

Comment (ed-brex) [reply Casks/threads.rb]: Got a response:
> We don’t currently provide a list/feed/appcast for the desktop version yet.... We are currently working on a “What’s new page” and the team confirmed they will look at adding a specific updates feed for the app versions as part of that work.
So not as of yet - but hopefully near future.

File: Casks/threads.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  appcast 'https://d111it5gwc8pcx.cloudfront.net/static/App.4a47906c52618800f007.js'
```
Comment (vitorgalvao) [inline Casks/threads.rb]: I doubt this URL will keep working, but let’s see.

**Pull request merged**